### PR TITLE
feat: Add KindInfo to Extension Upsert - BED-8764

### PR DESCRIPTION
**Pull request by LawsonWillard**
## Description

- Adds kindInfo to the Upsert Extension endpoint, allowing extension authors to add contextual information to specific kinds. 
- Adds unit and integration tests to cover the changeset.  

## Motivation and Context
Resolves: BED-8764

As an OpenGraph Extension Author, I want the /api/v2/extensions API to support KindInfo entries for node and relationship kinds so that I can keep those definitions in sync with my extension.

## How Has This Been Tested?

- Added new unit and integration tests to cover the feature 
- Tested locally to ensure kind info was added as expected

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

## Summary

* **New Features**
  * Added configurable “info” panels (title, position, markdown) for both node kinds and relationship kinds, keyed per entry.
  * Kind info is now fully reconciled during graph extension ingest (create/update/delete).
* **Bug Fixes**
  * Stricter server-side validation for panel keys, counts, positions, and markdown shape, with consistent API error mapping.
  * Kind identifiers are reliably returned/persisted; empty markdown content now defaults to the expected structure.
* **Documentation**
  * Updated OpenAPI schemas and endpoint examples to include the new info panels.
* **Tests**
  * Expanded unit and integration coverage for parsing, validation, and reconciliation edge cases.

### CODE REVIEW

File: cmd/api/src/api/v2/opengraphschema.go [hunk 1]
```diff
@@ -205,28 +265,22 @@ func convertGraphExtensionPayloadToGraphExtension(payload GraphExtensionPayload)
...
+
 		graphExtension.RelationshipKindsInput = append(graphExtension.RelationshipKindsInput,
 			model.RelationshipInput{
 				Name:          edgeKindPayload.Name,
 				Description:   edgeKindPayload.Description,
 				IsTraversable: edgeKindPayload.IsTraversable,
+				Info:          infoInputs,
 			})
 	}
-	/*
```
Comment (urangel) [inline cmd/api/src/api/v2/opengraphschema.go:218]: thanks for cleaning this up!

File: cmd/api/src/api/v2/opengraphschema.go [hunk 2]
```diff
@@ -179,8 +186,56 @@ func extractExtensionDataFromJSON(payload io.Reader) (GraphExtensionPayload, err
...
+			"markdown": markdown,
+		}
+
+		if contentBytes, err = json.Marshal(wrappedContent); err != nil {
+			return nil, fmt.Errorf("failed to wrap markdown content for info key %s: %w", key, err)
+		}
+
+		result = append(result, model.KindInfoInput{
+			InfoKey:  key,
+			Title:    payload.Title,
```
Comment (ykaiboussiSO) [inline cmd/api/src/api/v2/opengraphschema.go]: Should we include validation on `Title` key? Currently the API is returning `500` status on missing the key or passing an empty string.

Comment (urangel) [reply cmd/api/src/api/v2/opengraphschema.go]: Yeah I think we should either return a 400 with a descriptive error message if we want to have non-empty titles or allow empty titles and avoid the 500. Probably the former

Comment (LawsonWillard) [reply cmd/api/src/api/v2/opengraphschema.go]: Whoops, thought i had a validation check for title, definitely need that and I agree with @urangel that we should be returning a 400. 

Will get this added in a min

File: cmd/api/src/api/v2/opengraphschema.go (hunk 1 above)
Comment (LawsonWillard) [reply cmd/api/src/api/v2/opengraphschema.go:218]: I think im going to cut another ticket to get all the old property stuff removed, there's an unused table as well as db functions, but its a little much to include here.

File: cmd/api/src/model/graphschema.go [hunk 3]
```diff
@@ -45,17 +46,25 @@ var (
...
+
+	// entity panel duplicate errors, enforced at both the model (validation) and db (uniqueness constraint) layers
 	ErrKindInfoDuplicatePosition = errors.New("kind info position already in use for this kind")
 	ErrKindInfoDuplicateInfoKey  = errors.New("kind info key already in use for this kind")
+
+	// entity panel validation errors
+	ErrInvalidKindInfoTitle    = errors.New("invalid kind info title: must not be empty or whitespace-only")
+	ErrInvalidKindInfoKey      = errors.New("invalid kind info key: must match pattern ^[a-z0-9_-]{1,128}$")
+	ErrTooManyKindInfoEntries  = errors.New("too many kind info entries: maximum 100 allowed per kind")
+	ErrInvalidKindInfoPosition = errors.New("invalid kind info position: must be >= 1")
```
Comment (brandonshearin) [inline cmd/api/src/model/graphschema.go]: lets double check this -- should position start from 0 or 1?

**Review by urangel — APPROVED** [review]
Nice work!

File: schemas/open_graph/dog-park-schema.json [hunk 4]
```diff
@@ -12,36 +12,95 @@
...
+          "title": "Overview",
+          "position": 1,
+          "markdown": {
+            "content": "A **Dog** is a happy park attendee owned by a patron."
+          }
+        }
+      }
     }
   ],
   "properties": [],
```
Comment (kpowderly) [inline schemas/open_graph/dog-park-schema.json]: I know this isn't apart of your PR but properties in the upsert extension schema is dead now, right?

**Comment by kpowderly** [comment]
need a corresponding enterprise branch but looks good! excited to see the entity panel work for OG!!

**Review by kpowderly — APPROVED** [review]

File: schemas/open_graph/dog-park-schema.json (hunk 4 above)
Comment (LawsonWillard) [reply schemas/open_graph/dog-park-schema.json]: Yes! Thank you for reminding me, i add a commit to remove this in a sec

File: cmd/api/src/model/graphschema.go (hunk 3 above)
Comment (LawsonWillard) [reply cmd/api/src/model/graphschema.go]: Thanks for the callout, you're right, should be starting at 0. Will update this in a sec

File: cmd/api/src/api/v2/opengraphschema_internal_test.go [hunk 5]
```diff
@@ -145,10 +134,183 @@ func Test_convertGraphExtensionPayloadToGraphExtension(t *testing.T) {
...
+		},
+		{
+			name: "error_-_position_exceeds_int32",
+			input: map[string]KindInfoPayload{
+				"overflow": {
+					Title:    "Overflow",
+					Position: math.MaxInt32 + 1, // would silently wrap when cast to int32
+					Markdown: []byte(`{"content":"# Test"}`),
+				},
+			},
```
Comment (AD7ZJ) [inline cmd/api/src/api/v2/opengraphschema_internal_test.go]: Might be worth a test case for zero position too - I don't think it would throw an error as it is now, but it probably should?

**Review by AD7ZJ — APPROVED** [review]
One nit on the position validation, otherwise looks good to me!
